platform/darwin/test/MGLPredicateTests.mm

```diff
@@ -576,6 +576,81 @@ - (void)testComparisonExpressionArray {
+    }
+    {
+        NSArray *expected = @[@"all", @[@">=",  @[@"get", @"x"], @10], @[@">=", @100, @[@"get", @"x"]]];
+        NSExpression *limits = [NSExpression expressionForAggregate:@[[NSExpression expressionForConstantValue:@10], [NSExpression expressionForConstantValue:@100]]];
+        NSPredicate *predicate = [NSPredicate predicateWithFormat:@"x BETWEEN %@", limits];
+        XCTAssertEqualObjects([NSPredicate mgl_predicateWithJSONObject:expected], predicate);
+    }
+    {
+        NSArray *expected = @[@"all", @[@"==", @10, @[@"get", @"x"]], @[@"<=", @[@"get", @"x"], @100]];
+        NSPredicate *predicate = [NSPredicate predicateWithFormat:@"%@ == x && x <= %@", [NSExpression expressionForConstantValue:@10], [NSExpression expressionForConstantValue:@100]];
+        XCTAssertEqualObjects(predicate.mgl_jsonExpressionObject, expected);
+        XCTAssertEqualObjects([NSPredicate mgl_predicateWithJSONObject:expected], predicate);
+    }
+    {
+        NSArray *expected = @[@"match", @[@"id"], @6, @YES, @5, @YES, @4, @YES, @3, @YES, @NO];
+        NSPredicate *predicate = [NSPredicate predicateWithFormat:@"$mgl_featureIdentifier IN { 6, 5, 4, 3}"];
+        XCTAssertEqualObjects(predicate.mgl_jsonExpressionObject, expected);
+        NSString *mglMatch = @"MGL_MATCH($mgl_featureIdentifier, 6, YES, 5, YES, 4, YES, 3, YES, NO)";
+        NSPredicate *predicateAfter = [self matchPredicateWithFormat:mglMatch];
+        XCTAssertEqualObjects([NSPredicate mgl_predicateWithJSONObject:expected], predicateAfter);
+    }
+    {
+        NSArray *expected = @[@"!", @[@"match", @[@"get", @"x"], @6, @YES, @5, @YES, @4, @YES, @3, @YES, @NO]];
+        NSPredicate *predicate = [NSPredicate predicateWithFormat:@"NOT x IN { 6, 5, 4, 3}"];
+        XCTAssertEqualObjects(predicate.mgl_jsonExpressionObject, expected);
+    }
+    {
+        NSArray *expected = @[@"match", @[@"get", @"x"], @6, @YES, @5, @YES, @4, @YES, @3, @YES, @NO];
+        NSPredicate *predicate = [NSPredicate predicateWithFormat:@"{ 6, 5, 4, 3} CONTAINS x"];
+        XCTAssertEqualObjects(predicate.mgl_jsonExpressionObject, expected);
+        NSString *mglMatch = @"MGL_MATCH(x, 6, YES, 5, YES, 4, YES, 3, YES, NO)";
+        NSPredicate *predicateAfter = [self matchPredicateWithFormat:mglMatch];
+        XCTAssertEqualObjects([NSPredicate mgl_predicateWithJSONObject:expected], predicateAfter);
+    }
+    {
+        NSArray *expected = @[@"match", @[@"id"], @6, @YES, @5, @YES, @4, @YES, @3, @YES, @NO];
+        NSPredicate *predicate = [NSPredicate predicateWithFormat:@"{ 6, 5, 4, 3} CONTAINS $mgl_featureIdentifier"];
+        XCTAssertEqualObjects(predicate.mgl_jsonExpressionObject, expected);
+        NSString *mglMatch = @"MGL_MATCH($mgl_featureIdentifier, 6, YES, 5, YES, 4, YES, 3, YES, NO)";
+        NSPredicate *predicateAfter = [self matchPredicateWithFormat:mglMatch];
+        XCTAssertEqualObjects([NSPredicate mgl_predicateWithJSONObject:expected], predicateAfter);
```

Comment:
Add tests that convert `match` expressions to NSPredicates.

Reply:
What is the purpose of this test? comparing directly `match` expressions to `match` expression predicates will not work because we add the condition `... == YES`

Reply:
Right, what I mean is that we should test that behavior, to verify that https://github.com/mapbox/mapbox-gl-native/pull/11632#discussion_r181178469 is fixed.

Reply:
Looks like you already added them, never mind.